react/reducer/index.ts

```diff
@@ -95,10 +96,18 @@ function reducer(
 
       return {
         ...state,
+        loadingItem: false,
```

Comment:
It's not clear to me why you set this here. Who dispatches `SET_SELECTED_ITEM` and that's the guarantee will it always run after `SET_LOADING_ITEM`? Shouldn't it be safer to make another dispatch to `SET_LOADING_ITEM` setting it as false?

Reply:
The flow is:
1. user clicks on the SKU selector -> SET_LOADING_ITEM and changes the SKU query string (this behavior already exists)
2. In the ProductContext provider you can see when the query changes, the component re-renders and calls [useSelectedItemFromId](https://github.com/vtex-apps/product-context/blob/9767af0864bb9a6bdf45de5652172c0ede12c5f8/react/ProductContextProvider.tsx#L122) -> this hook dispatch the `SET_SELECTED_ITEM`

Reply:
Always when the product context receives new sku and will set the selected item, it will set the loadingItem to false.

Reply:
Got it! Thanks!